vortex-file/src/read/layouts/columnar.rs

```diff
@@ -303,6 +305,41 @@ impl LayoutReader for ColumnarLayoutReader {
             Ok(Some(BatchRead::ReadMore(messages)))
         }
     }
+
+    fn read_metadata(&self) -> VortexResult<MetadataRead> {
+        let mut in_progress_metadata = self
+            .in_progress_metadata
+            .write()
+            .unwrap_or_else(|e| vortex_panic!("lock is poisoned: {e}"));
+        let mut messages = Vec::default();
+
+        for (name, child_reader) in self.names.iter().zip(self.children.iter()) {
+            match child_reader.read_metadata()? {
+                MetadataRead::Batches(data) => {
+                    in_progress_metadata.insert(name.clone(), data[0].clone());
```

Comment:
we grab `data[0]` but what about the rest of the `data` vec? This code implicitly asserts that columnar layouts cannot have other columnar layouts as their children, we should just assert explicitly that `data` is of length 1

Reply:
We need to add some abstraction for stats. It could be just `names` and `list of arrays` since we need to handle this case, so stats will be recursive